### PR TITLE
closes tux3/qtox#1671 restarting after changing lang

### PR DESCRIPTION
**Pull request by agilob**

### CODE REVIEW

File: src/widget/form/settings/generalform.cpp [hunk 1]
```diff
@@ -210,6 +211,16 @@ void GeneralForm::onTranslationUpdated()
...
     Widget::getInstance()->setTranslation();
+    Settings::getInstance().save();
+    QMessageBox::StandardButton reply;
+      reply = QMessageBox::question(0,
+                                    tr("Do you want to restart now?"),
+                                    tr("You must qTox restart to change the language"),
+                                    QMessageBox::Yes|QMessageBox::No);
+      if (reply == QMessageBox::Yes) {
+          QProcess::startDetached(QApplication::applicationFilePath());
+          exit(12);
```
Comment (tux3) [inline src/widget/form/settings/generalform.cpp:222]: Why the nonzero exit status?

Comment (agilob) [inline src/widget/form/settings/generalform.cpp:222]: It doesn't restart with any other exit code, just closes and doesn't start again.

Comment (tux3) [inline src/widget/form/settings/generalform.cpp:222]: ...what?

That's obviously not right.

Comment (agilob) [inline src/widget/form/settings/generalform.cpp:222]: Could you try it? On my desktop any other code doesn't restart qtox

Comment (tux3) [inline src/widget/form/settings/generalform.cpp:222]: Yes, it works with any return code here.

The problem is not the return code. It's probably just the new instance racing with the current one. If it starts before we exit, it'll just maximize the current running instance (which is about to quit).
So we need to release the IPC's hold on the shared memory before starting the new process.

**Comment by zetok** [comment]
Wouldn't it be better to make language switch on runtime?

**Comment by tux3** [comment]
@zetok That's the ideal solution.

**Comment by TheLastProject** [comment]
@agilob "You must restart qTox" is correct grammar, your current "You must qTox restart" is wrong.

**Pull request closed without merging**